src/main/java/com/algorithmia/algorithmHandler/RequestHandler.java

```diff
@@ -0,0 +1,55 @@
+package com.algorithmia.algorithmHandler;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.util.Scanner;
+import org.apache.commons.codec.binary.Base64;
+
+
+final class RequestHandler<INPUT>
+{
+
+    private Scanner input;
+    private JsonParser parser;
+
+    RequestHandler(){
+        this.input = new Scanner(System.in);
+        this.parser = new JsonParser();
+    }
+
+    private Request<INPUT> CreateRequest(String contentType, JsonElement data) throws Exception {
+        if (contentType.equals("json")) {
+            if (data.isJsonArray()) {
+                JsonArray array = data.getAsJsonArray();
+                return new Request<>(contentType, array);
+            }else if(data.isJsonPrimitive() || data.isJsonNull()){
+                Object[] inputs = {data};
+                return new Request<>(contentType, inputs);
+            } else {
+                return new Request<>(contentType, data);
+            }
+        } else if (contentType.equals("text")) {
+            return new Request<>(contentType, data.getAsString());
+        } else if (contentType.equals("binary")) {
+            byte[] bytes = Base64.decodeBase64(data.getAsString());
+            return new Request<>(contentType, bytes);
+        } else {
+            throw new Exception("recieved an invalid content_type.");
```

Comment:
I'd strongly recommend that you not throw checked exceptions; instead, throw something in the `RuntimeException` hierarchy. Also, it would be valuable to the caller to indicate what the invalid content type was. How about `throw new RuntimeException("Cannot create a request with unrecognized content type " + contentType")`.